File: 백준/Gold/1300. K번째 수/K번째 수.cc
#include <iostream>
#include <algorithm>

using namespace std;

int main() {
	int n, k;
	cin >> n >> k;

	int left = 1, right = k, ans;
	while (left <= right) {
		long long cnt = 0;
		int mid = (left + right) / 2;
		for (int i = 1; i <= n; i++) cnt += min(mid / i, n);
		if (cnt < k) left = mid + 1;
		else ans = mid, right = mid - 1;
	}
	printf("%d", ans);
}

File: 백준/Gold/1300. K번째 수/K번째 수.py
def solution(N, k):
    low, high = 1, N * N

    while low < high:
        mid = (low + high) // 2
        cnt = sum(min(mid // i, N) for i in range(1, N + 1))

        if cnt < k:
            low = mid + 1
        else:
            high = mid

    return low


N = int(input())
k = int(input())
print(solution(N, k))